### PR TITLE
#375 add fontist for required font installation

### PR DESCRIPTION
**Pull request by CAMOBAP**
- #375

### CODE REVIEW

File: lib/asciidoctor/standoc/converter.rb [hunk 1]
```diff
@@ -75,7 +80,32 @@ def self.inherited(k)
...
+            fonts_manifest,
+            confirmation: options[:confirm_license] ? "yes" : "no"
+          )
+        rescue Fontist::Errors::LicensingError
+          log_type = options[:continue_without_fonts] ? :error : :fatal
+          Metanorma::Util.log("[fontinst] Error: License acceptance required to install a necessary font." \
+            "Accept required licenses with: `metanorma setup --agree-to-terms`.", log_type)
+
+        rescue Fontist::Errors::NonSupportedFontError
+          Metanorma::Util.log("[fontinst] The font `#{font}` is not yet supported.", :info)
```
[the remark this answers is not in the document]
Comment (ronaldtse) [reply lib/asciidoctor/standoc/converter.rb:107]: "fontist" not "fontinst"

File: lib/asciidoctor/standoc/converter.rb [hunk 2]
```diff
@@ -75,7 +80,32 @@ def self.inherited(k)
...
+        end
+
+        begin
+          Fontist::Manifest::Install.call(
+            fonts_manifest,
+            confirmation: options[:confirm_license] ? "yes" : "no"
+          )
+        rescue Fontist::Errors::LicensingError
+          log_type = options[:continue_without_fonts] ? :error : :fatal
+          Metanorma::Util.log("[fontinst] Error: License acceptance required to install a necessary font." \
```
[the remark this answers is not in the document]
Comment (ronaldtse) [reply lib/asciidoctor/standoc/converter.rb:103]: also, "fontist" not "fontinst"

File: lib/asciidoctor/standoc/converter.rb [hunk 3]
```diff
@@ -75,7 +80,32 @@ def self.inherited(k)
...
+        File.join(@libdir, "../../isodoc/html", file)
+      end
+
+      def fonts_manifest
+        File.join(@libdir, "fonts_manifest.yaml")
+      end
+
+      def install_fonts(options={})
+        if options[:no_install_fonts] || fonts_manifest.nil? || !File.exist?(fonts_manifest)
+          Metanorma::Util.log("[fontinst] Skip font installation process", :debug)
```
[the remark this answers is not in the document]
Comment (ronaldtse) [reply lib/asciidoctor/standoc/converter.rb:92]: "fontist" not "fontinst"

[the remark this answers is not in the document]
Comment (ronaldtse) [reply lib/asciidoctor/standoc/converter.rb:92]: Please output the reason why font installation is skipped. e.g. 
* "Skipping installation of fonts -- no extra fonts to be installed for this flavor."
* "Skipping installation of fonts -- `no_install_fonts` option enabled"

File: lib/asciidoctor/standoc/converter.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (ronaldtse) [reply lib/asciidoctor/standoc/converter.rb:103]: * "`confirm_license` option missing. You must accept font licenses to install fonts."
* "`continue_without_fonts` option enabled. Processing will continue without fonts installed".
* If `confirm_license` is not set and `continue_without_fonts` not set: "Aborting without proper fonts installed".

File: lib/asciidoctor/standoc/converter.rb [hunk 4]
```diff
@@ -75,7 +80,32 @@ def self.inherited(k)
...
+
+        begin
+          Fontist::Manifest::Install.call(
+            fonts_manifest,
+            confirmation: options[:confirm_license] ? "yes" : "no"
+          )
+        rescue Fontist::Errors::LicensingError
+          log_type = options[:continue_without_fonts] ? :error : :fatal
+          Metanorma::Util.log("[fontinst] Error: License acceptance required to install a necessary font." \
+            "Accept required licenses with: `metanorma setup --agree-to-terms`.", log_type)
```
[the remark this answers is not in the document]
Comment (ronaldtse) [reply lib/asciidoctor/standoc/converter.rb:104]: This is good.

File: lib/asciidoctor/standoc/converter.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (ronaldtse) [reply lib/asciidoctor/standoc/converter.rb:107]: Maybe add "Please file an issue at github.com/metanorma/metanorma-{flavor} to report this issue."?

**Review by ronaldtse — CHANGES_REQUESTED** [review]

**Review by opoudjis — APPROVED** [review]
Nice, and I see you reused my existing inheritance hook. Merge from master btw to update the relaton VCR cache, which is why rake is currently failing.

**Comment by CAMOBAP** [comment]
@ronaldtse could you please review again

**Review by ronaldtse — APPROVED** [review]

**Pull request merged**